src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx

```diff
@@ -0,0 +1,199 @@
+                    ) : currency ? (
+                      <CurrencyLogo
+                        size="20px"
+                        currency={currency}
+                        chainIdOverride={chainIdOverride}
+                        currencyWrapperSource={currencyWrapperSource}
+                      />
+                    ) : null}
+                    {pair ? (
+                      <StyledTokenName className="pair-name-container">
+                        {pair?.token0.symbol}/{pair?.token1.symbol}
+                      </StyledTokenName>
+                    ) : (
+                      <StyledTokenName className="token-symbol-container" active={Boolean(currency && currency.symbol)}>
+                        {(currency && currency.symbol && currency.symbol.length > 20
+                          ? currency.symbol.slice(0, 4) +
+                            '...' +
+                            currency.symbol.slice(currency.symbol.length - 5, currency.symbol.length)
+                          : currency?.symbol) || <div data-testid="select-token-button"> {t('select Token')}</div>}
+                      </StyledTokenName>
+                    )}
+                    {!disableCurrencySelect && (pair || currency) && <StyledDropDown selected={!!currency} />}
+                  </>
+                )}
+              </Aligner>
+            </CurrencySelect>
+          </InputRow>
+          <FiatRow>
+            <RowBetween>
+              {fiatValue && (
+                <FiatValueDetails
+                  fiatValue={fiatValue?.toFixed(2, { groupSeparator: ',' })}
+                  priceImpact={priceImpact}
+                  isFallback={isFallbackFiatValue}
+                />
+              )}
+              {account && (
+                <TYPE.body
+                  onClick={onMax}
+                  fontWeight="600"
+                  fontSize="10px"
+                  lineHeight="13px"
+                  letterSpacing="0.08em"
+                  style={{
+                    display: 'inline',
+                    marginLeft: 'auto',
+                    cursor:
+                      !hideBalance && !!(currency || pair) && (balance || selectedCurrencyBalance) ? 'pointer' : 'auto',
+                  }}
+                >
+                  <UppercaseHelper>
+                    {!hideBalance && !!(currency || pair) && (balance || selectedCurrencyBalance) && (
+                      <>
+                        {customBalanceText ?? t('balance')}
+                        <TYPE.small as="span" fontWeight="600" color="text3" style={{ textDecoration: 'underline' }}>
+                          {limitNumberOfDecimalPlaces(balance || selectedCurrencyBalance) || '0'}
+                        </TYPE.small>
+                      </>
+                    )}
+                  </UppercaseHelper>
+                </TYPE.body>
+              )}
```

Comment:
Split logic to sub components if possible

Reply:
Fixed. You can take a look again :)